### ISSUE TITLE
Search is waiting for a background job, but no thread has any matching activity

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-jdt/eclipse.jdt.core#41]
After jumping through some more maven hoops, I managed to get stacktraces from the freeze: [freeze-jstack.txt](https://github.com/eclipse-jdt/eclipse.jdt.core/files/8562807/freeze-jstack.txt)

two interesting portions extracted here:
```
"main" #1 prio=5 os_prio=0 cpu=32921,05ms elapsed=181,43s tid=0x00007fdd680261e0 nid=0x35536 in Object.wait()  [0x00007fdd6fb78000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(java.base@17/Native Method)
	- waiting on <no object reference available>
	at org.eclipse.jdt.internal.core.search.processing.JobManager.performConcurrentJob(JobManager.java:284)
	- locked <0x00000000c05bbbe0> (a org.eclipse.jdt.internal.core.search.indexing.IndexManager)
	at org.eclipse.jdt.internal.core.search.BasicSearchEngine.searchAllTypeNames(BasicSearchEngine.java:1923)
	at org.eclipse.jdt.internal.core.search.BasicSearchEngine.searchAllTypeNames(BasicSearchEngine.java:1751)
	at org.eclipse.jdt.core.search.SearchEngine.searchAllTypeNames(SearchEngine.java:1100)
	at org.eclipse.jdt.core.tests.model.AbstractJavaModelTests.waitUntilIndexesReady(AbstractJavaModelTests.java:3967)
	at org.eclipse.jdt.core.tests.dom.ASTConverterBugsTest.setUpSuite(ASTConverterBugsTest.java:41)
...
"Java indexing" #31 daemon prio=5 os_prio=0 cpu=871,25ms elapsed=178,36s tid=0x00007fdd68640200 nid=0x355bb in Object.wait()  [0x00007fdd402f9000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(java.base@17/Native Method)
	- waiting on <no object reference available>
	at java.lang.Object.wait(java.base@17/Object.java:338)
	at org.eclipse.jdt.internal.core.search.processing.JobManager.run(JobManager.java:419)
	- locked <0x00000000c05bbbe0> (a org.eclipse.jdt.internal.core.search.indexing.IndexManager)
	at java.lang.Thread.run(java.base@17/Thread.java:833)
```

I read this as: search is waiting for a background job, but no thread has any matching activity.

_Originally posted by @stephan-herrmann in https://github.com/eclipse-jdt/eclipse.jdt.core/issues/38#issuecomment-1109697373_

**Comment by stephan-herrmann** [comment]
In https://github.com/eclipse-jdt/eclipse.jdt.core/pull/47#issuecomment-1110202356 I noticed, that the recent freeze in some pull requests depens on execution order within org.eclipse.jdt.core.tests.model.

This sequence reproducibly freezes:
1. org.eclipse.jdt.core.tests.RunFormatterTests
2. org.eclipse.jdt.core.tests.dom.RunAllTestsTracing

Additionally, I managed to see the freeze in the debugger, which shows:
* Thread "main" waits for completion of an AddJarFileToIndex "[indexing /home/git/eclipse.jdt.core/org.eclipse.jdt.core.tests.model/target/work/converterJclMin.jar]"
* Thread "Java indexing" is stuck in `this.wait()`
  * apparently the last `currentJob()` didn't find the queued AddJarFileToIndex
  * apparently that happened because `this.enableCount` is `-116`, which reminds me of the long cascades of `ENABLING  background indexing` then `DISABLING  background indexing` in https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/view/change-requests/job/PR-28/10/consoleText

**Comment by stephan-herrmann** [comment]
@iloveeclipse  the test-variant of this bug is caused by https://git.eclipse.org/c/jdt/eclipse.jdt.core.git/commit/?id=09687d64ecd9d0744a335c284c64c99f491d7e64 where you unconditionally disable even a disabled indexer, so that next `enable()` doesn't have the desired effect.

In #48 I'm testing a fix right in `org.eclipse.jdt.core.tests.junit.extension.TestCase.setUp()`, but on the long run we could consider protecting `org.eclipse.jdt.internal.core.search.processing.JobManager.disable()` from underrun, to ensure that **every** `enable()` actually enables.

Actually, all non-test callers of `disable()` seem to be well-balanced, at first glance that is. We should go through linked bugs to see if any stray calls to `disable()` can be found.

**Comment by jukzi** [comment]
happend also in https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-26/13/console
```

13:25:16  [2022-05-09 11:25:16 +0000] org.eclipse.jdt.core.tests.model.CompletionTests18#testUnspecifiedReference()
...
13:30:23  Possible frozen test case
13:30:23  "main": TIMED_WAITING
13:30:23      java.base@17.0.1/java.lang.Object.wait(Native Method)
13:30:23      org.eclipse.jdt.internal.core.search.processing.JobManager.performConcurrentJob(JobManager.java:284)
13:30:23      org.eclipse.jdt.internal.core.search.indexing.IndexManager.waitForIndex(IndexManager.java:176)
13:30:23      org.eclipse.jdt.core.tests.model.AbstractJavaModelTests.waitUntilIndexesReady(AbstractJavaModelTests.java:4008)

...

13:30:23  "Java indexing": WAITING
13:30:23      java.base@17.0.1/java.lang.Object.wait(Native Method)
13:30:23      java.base@17.0.1/java.lang.Object.wait(Object.java:338)
13:30:23      org.eclipse.jdt.internal.core.search.processing.JobManager.run(JobManager.java:419)
13:30:23      java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
```

**Comment by iloveeclipse** [comment]
Most recent case, same stack: https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-229/3/console
```
Possible frozen test case
"main": TIMED_WAITING
    java.base@17.0.1/java.lang.Object.wait(Native Method)
    org.eclipse.jdt.internal.core.search.processing.JobManager.performConcurrentJob(JobManager.java:284)
    org.eclipse.jdt.internal.core.search.indexing.IndexManager.waitForIndex(IndexManager.java:176)
    org.eclipse.jdt.core.tests.model.AbstractJavaModelTests.waitUntilIndexesReady(AbstractJavaModelTests.java:4008)
    org.eclipse.jdt.core.tests.model.AbstractJavaModelCompletionTests.setUpSuite(AbstractJavaModelCompletionTests.java:175)
    org.eclipse.jdt.core.tests.model.CompletionTests9.setUpSuite(CompletionTests9.java:49)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.runTest(SuiteOfTestCases.java:105)
    junit.framework.TestSuite.run(TestSuite.java:236)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.superRun(SuiteOfTestCases.java:98)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite$1.protect(SuiteOfTestCases.java:86)
    junit.framework.TestResult.runProtected(TestResult.java:142)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.run(SuiteOfTestCases.java:95)
    junit.framework.TestSuite.runTest(TestSuite.java:241)
    junit.framework.TestSuite.run(TestSuite.java:236)
    junit.framework.TestSuite.runTest(TestSuite.java:241)
    junit.framework.TestSuite.run(TestSuite.java:236)
    org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:90)
    org.junit.runners.Suite.runChild(Suite.java:128)
    org.eclipse.test.TracingSuite.runChild(TracingSuite.java:290)

"Java indexing": WAITING
    java.base@17.0.1/java.lang.Object.wait(Native Method)
    java.base@17.0.1/java.lang.Object.wait(Object.java:338)
    org.eclipse.jdt.internal.core.search.processing.JobManager.run(JobManager.java:419)
    java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
```

**Comment by iloveeclipse** [comment]
Still there, see https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-245/2/console.

Stack is slightly different now:
```
"Java indexing": WAITING
    java.base@17.0.1/java.lang.Object.wait(Native Method)
    java.base@17.0.1/java.lang.Object.wait(Object.java:338)
    org.eclipse.jdt.internal.core.search.processing.JobManager.run(JobManager.java:452)
    java.base@17.0.1/java.lang.Thread.run(Thread.java:833)
```

 ```
"main": TIMED_WAITING
    java.base@17.0.1/java.lang.Object.wait(Native Method)
    org.eclipse.jdt.internal.core.search.processing.JobManager.performConcurrentJob(JobManager.java:311)
    org.eclipse.jdt.internal.core.search.indexing.IndexManager.waitForIndex(IndexManager.java:176)
    org.eclipse.jdt.core.tests.model.AbstractJavaModelTests.waitUntilIndexesReady(AbstractJavaModelTests.java:4008)
    org.eclipse.jdt.core.tests.model.AbstractJavaModelCompletionTests.setUpSuite(AbstractJavaModelCompletionTests.java:175)
    org.eclipse.jdt.core.tests.model.CompletionTests9.setUpSuite(CompletionTests9.java:49)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.runTest(SuiteOfTestCases.java:105)
    junit.framework.TestSuite.run(TestSuite.java:236)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.superRun(SuiteOfTestCases.java:98)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite$1.protect(SuiteOfTestCases.java:86)
    junit.framework.TestResult.runProtected(TestResult.java:142)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.run(SuiteOfTestCases.java:95)
    junit.framework.TestSuite.runTest(TestSuite.java:241)
    junit.framework.TestSuite.run(TestSuite.java:236)
    junit.framework.TestSuite.runTest(TestSuite.java:241)
    junit.framework.TestSuite.run(TestSuite.java:236)
    org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:90)
```

**Comment by iloveeclipse** [comment]
Still there on master:

https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-662/2/console

```
Possible frozen test case
"main": TIMED_WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    org.eclipse.jdt.internal.core.search.processing.JobManager.performConcurrentJob(JobManager.java:311)
    org.eclipse.jdt.internal.core.search.indexing.IndexManager.waitForIndex(IndexManager.java:176)
    org.eclipse.jdt.core.tests.model.AbstractJavaModelTests.waitUntilIndexesReady(AbstractJavaModelTests.java:4021)
    org.eclipse.jdt.core.tests.model.AbstractJavaModelCompletionTests.setUpSuite(AbstractJavaModelCompletionTests.java:175)
    org.eclipse.jdt.core.tests.model.CompletionTests9.setUpSuite(CompletionTests9.java:49)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.runTest(SuiteOfTestCases.java:105)
    junit.framework.TestSuite.run(TestSuite.java:236)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.superRun(SuiteOfTestCases.java:98)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite$1.protect(SuiteOfTestCases.java:86)
    junit.framework.TestResult.runProtected(TestResult.java:142)
    org.eclipse.jdt.core.tests.model.SuiteOfTestCases$Suite.run(SuiteOfTestCases.java:95)
    junit.framework.TestSuite.runTest(TestSuite.java:241)
    junit.framework.TestSuite.run(TestSuite.java:236)
    junit.framework.TestSuite.runTest(TestSuite.java:241)
    junit.framework.TestSuite.run(TestSuite.java:236)
    org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:90)
    org.junit.runners.Suite.runChild(Suite.java:128)
    org.eclipse.test.TracingSuite.runChild(TracingSuite.java:290)
    org.junit.runners.Suite.runChild(Suite.java:27)
    org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
    org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
    org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
    org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
    org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
    org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
    org.junit.runners.ParentRunner.run(ParentRunner.java:413)
    org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
    org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
    org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
    org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
    java.base@19/java.lang.invoke.LambdaForm$DMH/0x00000008011b0800.invokeVirtual(LambdaForm$DMH)
    java.base@19/java.lang.invoke.LambdaForm$MH/0x0000000801001800.invoke(LambdaForm$MH)
    java.base@19/java.lang.invoke.Invokers$Holder.invokeExact_MT(Invokers$Holder)
    java.base@19/jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:155)
    java.base@19/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    java.base@19/java.lang.reflect.Method.invoke(Method.java:578)
    org.apache.maven.surefire.api.util.ReflectionUtils.invokeMethodWithArray2(ReflectionUtils.java:167)
    org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:161)
    org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:84)
    org.eclipse.tycho.surefire.osgibooter.OsgiSurefireBooter.run(OsgiSurefireBooter.java:142)
    org.eclipse.tycho.surefire.osgibooter.HeadlessTestApplication.start(HeadlessTestApplication.java:29)
    org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
    org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
    org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
    org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
    java.base@19/java.lang.invoke.DirectMethodHandle$Holder.invokeStatic(DirectMethodHandle$Holder)
    java.base@19/java.lang.invoke.LambdaForm$MH/0x0000000801006800.invoke(LambdaForm$MH)
    java.base@19/java.lang.invoke.LambdaForm$MH/0x0000000801006c00.invokeExact_MT(LambdaForm$MH)
    java.base@19/jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:156)
    java.base@19/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    java.base@19/java.lang.reflect.Method.invoke(Method.java:578)
    app//org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
    app//org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
    app//org.eclipse.equinox.launcher.Main.run(Main.java:1467)
    app//org.eclipse.equinox.launcher.Main.main(Main.java:1440)

"Reference Handler": RUNNABLE
    java.base@19/java.lang.ref.Reference.waitForReferencePendingList(Native Method)
    java.base@19/java.lang.ref.Reference.processPendingReferences(Reference.java:245)
    java.base@19/java.lang.ref.Reference$ReferenceHandler.run(Reference.java:207)

"Finalizer": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    java.base@19/java.lang.ref.NativeReferenceQueue.await(NativeReferenceQueue.java:48)
    java.base@19/java.lang.ref.ReferenceQueue.remove0(ReferenceQueue.java:158)
    java.base@19/java.lang.ref.NativeReferenceQueue.remove(NativeReferenceQueue.java:89)
    java.base@19/java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:173)

"Signal Dispatcher": RUNNABLE

"Common-Cleaner": TIMED_WAITING
    java.base@19/jdk.internal.misc.Unsafe.park(Native Method)
    java.base@19/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:269)
    java.base@19/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1759)
    java.base@19/java.lang.ref.ReferenceQueue.await(ReferenceQueue.java:71)
    java.base@19/java.lang.ref.ReferenceQueue.remove0(ReferenceQueue.java:143)
    java.base@19/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:218)
    java.base@19/jdk.internal.ref.CleanerImpl.run(CleanerImpl.java:140)
    java.base@19/java.lang.Thread.run(Thread.java:1589)
    java.base@19/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186)

"Notification Thread": RUNNABLE

"Active Thread: Equinox Container: ae6f0de6-a383-4b16-8129-cb161c3e8161": TIMED_WAITING
    java.base@19/jdk.internal.misc.Unsafe.park(Native Method)
    java.base@19/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:269)
    java.base@19/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1674)
    java.base@19/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1182)
    java.base@19/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:899)
    java.base@19/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1070)
    java.base@19/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
    java.base@19/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    java.base@19/java.lang.Thread.run(Thread.java:1589)

"Framework Event Dispatcher: Equinox Container: ae6f0de6-a383-4b16-8129-cb161c3e8161": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"Start Level: Equinox Container: ae6f0de6-a383-4b16-8129-cb161c3e8161": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"Bundle File Closer": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"Refresh Thread: Equinox Container: ae6f0de6-a383-4b16-8129-cb161c3e8161": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
    org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:341)

"SCR Component Actor": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    org.apache.felix.scr.impl.ComponentActorThread.run(ComponentActorThread.java:83)
    java.base@19/java.lang.Thread.run(Thread.java:1589)

"EMF Reference Cleaner": WAITING
    java.base@19/jdk.internal.misc.Unsafe.park(Native Method)
    java.base@19/java.util.concurrent.locks.LockSupport.park(LockSupport.java:371)
    java.base@19/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
    java.base@19/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3744)
    java.base@19/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3689)
    java.base@19/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1625)
    java.base@19/java.lang.ref.ReferenceQueue.await(ReferenceQueue.java:67)
    java.base@19/java.lang.ref.ReferenceQueue.remove0(ReferenceQueue.java:158)
    java.base@19/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:234)
    org.eclipse.emf.common.util.CommonUtil$1ReferenceClearingQueuePollingThread.run(CommonUtil.java:70)

"Worker-JM": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    org.eclipse.core.internal.jobs.InternalWorker.run(InternalWorker.java:61)

"Worker-0: <unnamed job: org.eclipse.core.runtime.jobs.Job$2>": RUNNABLE
    java.management@19/sun.management.ThreadImpl.getThreadInfo1(Native Method)
    java.management@19/sun.management.ThreadImpl.getThreadInfo(ThreadImpl.java:192)
    org.eclipse.jdt.core.tests.model.FreezeMonitor$1.run(FreezeMonitor.java:41)
    org.eclipse.core.runtime.jobs.Job$2.run(Job.java:187)
    org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

"Timer-1": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    java.base@19/java.util.TimerThread.mainLoop(Timer.java:537)
    java.base@19/java.util.TimerThread.run(Timer.java:516)

"Worker-11": TIMED_WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:200)
    org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:242)
    org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)

"Java indexing": WAITING
    java.base@19/java.lang.Object.wait0(Native Method)
    java.base@19/java.lang.Object.wait(Object.java:366)
    java.base@19/java.lang.Object.wait(Object.java:339)
    org.eclipse.jdt.internal.core.search.processing.JobManager.run(JobManager.java:452)
    java.base@19/java.lang.Thread.run(Thread.java:1589)
```

**Comment by jukzi** [comment]
> Still there on master:

i proposed a patch .... there are synchonizes missing

**Comment by iloveeclipse** [comment]
> i proposed a patch .... there are synchonizes missing

OK, I must have missed that: https://github.com/eclipse-jdt/eclipse.jdt.core/pull/69

Will check.

**Comment by iloveeclipse** [comment]
Quick update: the issue is closed because we hope with #69 that fixes some of existing MT inconsistencies in the related code we shouldn't see the hangs in tests.

Please reopen if the problem reappears again & attach updates thread dumps!
